core/spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/PollableSourceTests.java

```diff
@@ -57,9 +62,66 @@ void test() {
 			assertThat(pollerMetadata.getTrigger()).isInstanceOf(CronTrigger.class);
 			assertThat(TestUtils.getPropertyValue(pollerMetadata.getTrigger(), "expression.expression"))
 				.isEqualTo("*/2 * * * * *");
+			assertThat(pollerMetadata.getMaxMessagesPerPoll()).isEqualTo(4);
 		}
 	}
 
+	@Test
+	void testPollerDefaultFixedDelay() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(TestChannelBinderConfiguration
+			.getCompleteConfiguration(PollableAppSampleConfiguration.class))
+			.web(WebApplicationType.NONE).run()) {
+
+			PollerMetadata pollerMetadata = context.getBean(PollerMetadata.class);
+			assertThat(pollerMetadata.getTrigger()).isInstanceOf(PeriodicTrigger.class);
+			assertThat(TestUtils.getPropertyValue(pollerMetadata.getTrigger(), "fixedRate")).isEqualTo(false);
```

Comment:
Not sure if you knew about this or not, but AssertJ has really great support for "digging at fields" via `assertThat(<thing>).hasFieldOrPropertyWithValue(`.

Reply:
Yep! Know that one.
It didn't work for me in some scenario and then I just keep forgetting to give it a second chance 😄